planet_explorer/tests/test_daily_imagery.py

```diff
@@ -391,3 +400,422 @@ def test_search_wrong_aoi(qtbot, logged_in_explorer_dock_widget, qgis_debug_enab
+                    ]
+                ],
+            ],
+            True,
+        ),
+        pytest.param(
+            TEST_MULTIPOLY,
+            [
+                [
+                    [
+                        [19.854093, -32.914008],
+                        [19.115666, -33.119873],
+                        [19.804865, -33.36154],
+                        [20.400082, -33.00799],
+                        [20.042057, -32.694718],
+                        [19.854093, -32.914008],
+                    ]
+                ],
+                [
+                    [
+                        [18.063967, -32.824502],
+                        [17.996837, -33.173577],
+                        [18.390665, -33.231756],
+                        [18.592054, -32.846879],
+                        [18.242979, -32.694718],
+                        [18.063967, -32.824502],
+                    ]
+                ],
+                [
+                    [
+                        [19.795914, -33.603207],
+                        [19.80039, -33.777745],
+                        [20.149464, -33.840399],
+                        [20.221069, -33.459997],
+                        [19.921223, -33.406293],
+                        [19.795914, -33.603207],
+                    ]
+                ],
+                [
+                    [
+                        [18.784493, -33.204904],
+                        [18.650233, -33.406293],
+                        [19.053012, -33.603207],
+                        [19.563198, -33.580831],
+                        [19.267827, -33.272034],
+                        [18.829246, -33.066169],
+                        [18.784493, -33.204904],
+                    ]
+                ],
+                [
+                    [
+                        [18.866631, -32.117031],
+                        [18.615466, -32.368195],
+                        [19.083323, -32.535639],
+                        [19.309864, -32.117031],
+                        [19.093172, -31.924963],
+                        [18.866631, -32.117031],
+                    ]
+                ],
+                [
+                    [
+                        [19.565953, -32.107181],
+                        [19.462532, -32.328797],
+                        [20.102757, -32.491316],
+                        [20.112606, -31.979136],
+                        [19.738321, -31.905264],
+                        [19.565953, -32.107181],
+                    ]
+                ],
+            ],
+            False,
+        ),
+    ],
+)
+def test_aoi_from_multiple_polygons(layer_dir, expected_coordinates, perform_selection):
+    """Tests the filter for the AOI read from no selection and a
+    selection on a layer loaded in QGIS. AOI calculated from each polygon.
+    """
+    aoi_filter = PlanetAOIFilter()
+    layer = QgsVectorLayer(layer_dir, "")
+    QgsProject.instance().addMapLayer(layer)
+
+    iface.setActiveLayer(layer)
```

Comment:
I believe here we should use the fixture `pe_qgis_iface` rather than the `iface` you've imported in order to avoid the error in the pipeline:
```
>       iface.setActiveLayer(layer)
E       AttributeError: 'NoneType' object has no attribute 'setActiveLayer'
```